pvlib/tests/test_solarposition.py

```diff
@@ -579,19 +579,20 @@ def test_declination():
 def test_analytical_zenith():
     times = pd.date_range(start="1/1/2015 0:00", end="12/31/2015 23:00",
                           freq="h").tz_localize('Etc/GMT+8')
+    times_utc = times.tz_convert('UTC')
```

Comment:
I don't see why this change is needed.  Shouldn't we continue to test these functions with local time zones?  

Same for `test_analytical_azimuth` and `test_sun_rise_set_transit_geometric`.

Reply:
It was added because the following `solarposition.equation_of_time_*` and `solarposition.declination_*` appearing in all of `test_analytical_zenith`, `test_analytical_azimuth`, `test_sun_rise_set_transit_geometric` require to be given a day of year in the same way with `irradiance.get_extra_radiation` and `clearsky.lookup_linke_turbidity` . Otherwise, similar inconsistencies arise since a day of year is local and varies from local timezones as well as a local midnight, while neither an equation of time nor a declination is local but global at all times. In fact, they seem to have been calculated based on Universal Time, UT, or UT1, nearly the same with UTC: https://articles.adsabs.harvard.edu//full/1989MNRAS.238.1529H/0001531.000.html .

Reply:
It's a good point that EoT and declination are global and thus have the dayofyear locality problem.  However, doesn't that mean that we should modify `solarposition.equation_of_time_*` and `solarposition.declination_*` to use `_pandas_to_utc` internally (rather than externally as with this test) as you have done with other functions?

Reply:
If they were to be modified, their function parameters `dayofyear` should be redefined first to accept datetimes, which differs from the internal changes to `irradiance.get_extra_radiation` and `clearsky.lookup_linke_turbidity` that allows `pandas.DatetimeIndex` for function arguments. It is true that I could replace the input `dayofyear` into datetimes but there was a reason that I eventually did not, which is also why I failed to modify `tools.localize_to_utc` but instead created a similar `_pandas_to_utc(pd_object)` in https://github.com/pvlib/pvlib-python/pull/2055#discussion_r1626330929, and also why I reverted a modified version of `_datetimelike_scalar_to_datetimeindex` in [#2055 (comment)](https://github.com/pvlib/pvlib-python/pull/2055/files/6f98df71f12e87a2d2abe729d8a4b74ce35e5884#r1626346483). Those functions a ereven internal functions whose names are prefixed with an underscore unlike `solarposition.equation_of_time_*` and `solarposition.declination_*`. In other words, if I changed the function,
> some of the previous anonymous users of it would get confused.
> this will be a surprise if this function gets used for something else.

Reply:
I see now.  I overlooked that `solarposition.equation_of_time_*` and `solarposition.declination_*` took `dayofyear` (not datetimes) as inputs.  Of course you are correct that it makes no sense to modify them.  Thank you for the explanation, and I apologize for the useless question!

Reply:
That's some serious downplaying! It was very thoughtful and that is why it was tried and discussed earlier. Thank you for reviewing my humble opinion!